code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -287,13 +289,17 @@
 	update_button_icon()
 
 /datum/action/xeno_action/activable/secrete_resin/use_ability(atom/A)
+	//xeno owner of this ability
 	var/mob/living/carbon/xenomorph/xowner = owner
-	if(SSmonitor.gamestate == SHUTTERS_CLOSED && CHECK_BITFIELD(SSticker.mode.flags_round_type, MODE_ALLOW_XENO_QUICKBUILD) && SSresinshaping.active)
+	var/turf/selectedturf = get_turf(xowner)
+	if(selectedturf.get_build_prereqs() || SSmonitor.gamestate != SHUTTERS_CLOSED)
+		if(get_dist(owner, A) > xowner.xeno_caste.resin_max_range) //Maximum range is defined in the castedatum with resin_max_range, defaults to 0
+			build_resin(get_turf(owner))
+		else
+			build_resin(get_turf(A))
+		return
+	else if(SSmonitor.gamestate == SHUTTERS_CLOSED && CHECK_BITFIELD(SSticker.mode.flags_round_type, MODE_ALLOW_XENO_QUICKBUILD) && SSresinshaping.active)
 		preshutter_build_resin(A)
```

Comment:
What's happening with this proc?
This is getting muddled up from what it was originally doing.

Its checking get_build_prereqs() for every build for the entire game no matter what, which is highly undesirable.
We should only be checking get_build_prereqs() if needed. i.e. if gamestate == SHUTTERS_CLOSED, we check. If its not SHUTTERS_CLOSED, we just build normally.